db/db_impl.cc

```diff
@@ -618,12 +618,12 @@ static void DumpMallocStats(std::string* stats) {
 #endif  // !ROCKSDB_LITE
 
 void DBImpl::MaybeDumpStats() {
-  if (immutable_db_options_.stats_dump_period_sec == 0) return;
+  if (mutable_db_options_.stats_dump_period_sec == 0) return;
```

Comment:
mutable_db_options_ is guarded by DBImpl::mutex_ but MaybeDumpStats() is called without holding mutex. So there will be race condition on the option if another thread call SetDBOptions. You need to lock the mutex, get the value of stats_dump_period_sec, and unlock mutex to finish the rest of work here.